File: server/src/main/java/com/foss/server/dao/MatchRepository.java
package com.foss.server.dao;

import com.foss.server.domain.match.Match;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import java.util.List;

public interface MatchRepository extends MongoRepository<Match, String>, MatchCustomRepository {
    @Query("{ 'validation': true, 'matchType': { '$in': ?0 }, $or: [ { 'team1': ?1 }, { 'team2': ?2 } ] }")
    Page<Match> findAllMatches(List<Integer> matchTypes, String ouid1, String ouid2, Pageable pageable);
}

File: android/domain/src/main/java/com/foss/foss/repository/MatchRepository.kt
package com.foss.foss.repository

import com.foss.foss.model.Match
import com.foss.foss.model.MatchType

interface MatchRepository {

    suspend fun requestRefresh(nickname: String)

    suspend fun fetchMatches(nickname: String, matchType: MatchType): List<Match>
}
